src/main/java/duke/TaskHandler.java

```diff
@@ -0,0 +1,147 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+import duke.exceptions.FileIoException;
+import duke.exceptions.IndexOutOfRangeException;
+import duke.exceptions.TaskDoneException;
+import duke.tasks.DeadlineTask;
+import duke.tasks.EventTask;
+import duke.tasks.Task;
+import duke.tasks.TodoTask;
+
+/**
+ * Represents a task handler that handles the tasks
+ */
+public class TaskHandler {
+    private ArrayList<Task> tasks;
+
+    public TaskHandler() {
+        this.tasks = new ArrayList<>();
+    }
+
+    public void clearTaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    public ArrayList<Task> getTaskList() {
+        return this.tasks;
+    }
+
+    public int getLength() {
+        return this.tasks.size();
+    }
+
+    public void addTodoTask(String taskName) {
+        tasks.add(new TodoTask(taskName));
+    }
+
+    public void addDeadlineTask(String taskName, LocalDate deadline) {
+        tasks.add(new DeadlineTask(taskName, deadline));
+    }
+
+    public void addEventTask(String taskName, LocalDateTime startTime, LocalDateTime endTime) {
+        tasks.add(new EventTask(taskName, startTime, endTime));
+    }
+
+    /**
+     * Returns the Task that is done
+     *
+     * @return A Task
+     */
+    public Task taskIsDone(int index) throws IndexOutOfRangeException, TaskDoneException {
```

Comment:
Perhaps this method name could be clearer if a verb was used. E.g. `markTaskAsDone`